jetty-core/jetty-client/src/main/java/org/eclipse/jetty/client/Response.java

```diff
@@ -188,7 +188,10 @@ interface AsyncContentListener extends ContentSourceListener
         @Override
         default void onContentSource(Response response, Content.Source contentSource)
         {
-            Runnable demandCallback = Invocable.from(Invocable.InvocationType.NON_BLOCKING, () -> onContentSource(response, contentSource));
+            // Ask the InvocationType to the Content.Source, where applications
+            // have set it there using Content.Source.demand(Runnable).
+            Invocable.InvocationType invocationType = Invocable.getInvocationType(contentSource);
+            Runnable demandCallback = Invocable.from(invocationType, () -> onContentSource(response, contentSource));
```

Comment:
I think we can improve `Invocable.from` so that for any `Runnable` passed in that is not `Invocable`, it will not wrap with a `ReadyTask` if the type is `BLOCKING`.